docs_src/manuals/user_manual/storages.hrst

```diff
@@ -1,371 +1,374 @@
-
-* ``void reset()``: reset the data_store. maybe deallocates memory.
-* ``bool valid() const``: check if underlying storage info and storage is valid.
-* ``const array<uint_t, ndims> &total_lengths() const``: return the array of total number of data points in each
-  direction
-* ``const array<uint_t, ndims> &strides() const``: return the array of (aligned) strides.
-* ``void sync() const``: synchronize the copies on the host and the target.
-* ``reactivate_target_write_views``: re-enables read-write device views (see `Data View`_)
-* ``reactivate_host_write_views``: re-enabled read-write host views (see `Data View`_)
-* ``std::string const &name() const``: retrieve the name of the storage.
-
-.. note::
-
- The :term:`Data Store` cannot be used to modify or access the data.
- In order to do so we use the view concept, which is explained next.
-
-.. _data-view:
-
---------------
-Data View
---------------
-
-The view is a lightweight object to access and modify the data stored in a ``data_store``.  Views can be generated for
-both host and target. Views can be created in read only or in read-write mode. The read-only views can be beneficial
-because they don't trigger synchronizations with the device. These functions are used to create views:
+^^^^^^^^^^^^^^^^^^^
+Data Store Synopsis
+^^^^^^^^^^^^^^^^^^^
 
 .. code-block:: gridtools
 
-    auto host_view = make_host_view(data_store);
-    auto host_view_ro = make_host_view<access_mode::read_only>(data_store);
-    auto target_view = make_target_view(data_store);
-    auto target_view_ro = make_target_view<access_mode::read_only>(data_store);
-
-A read-write view requests exclusive ownership on the data store, meaning that a read-write view on the host puts the
-data store in a state such that no view (neither read-only nor read-write) can be created or used on target.
+ template</* Implementation defined parameters */>
+ class data_store {
+   public:
+     static constexpr size_t ndims; /* Dimensionality */
+     using layout_t = /* Instantiation of gridtools::layout_map. */;
+     using data_t = /* Type of the element. */;
+     // The following invariant is held: for any data_store instancies that have
+     // the same kind_t the strides are also the same.
+     using kind_t = /* A type that identifies the strides set. */;
+
+     // Data store arbitrary label. Mainly for debugging.
+     std::string const &name() const;
+     // The sizes of the data store in each dimension.
+     array<unsigned, ndims> const &lengths() const;
+     // The strides of the data store in each dimension.
+     array<unsigned, ndims> const &strides() const;
+     // 1D length of the data store expressed in number of elements.
+     // Namely it is a pointer difference between the last and the first element minus one.
+     unsigned length() const;
+
+     // Supplementary object that holds lengths and strides.
+     storage::info<ndims> const &info() const;
```

Comment:
It is used internally for the views creation. 
Hypothetically it could be useful for the end user as well -- it also contains `index` method that maps N-dimensional index to linear offset.
I can remove it from the documentation is you feel such.
Alternatively I can get rid of `lengths()`, `strides()` and `length()` methods for a data store and view and force the using of `info()`